linkerd/transport-header/fuzz/fuzz_targets/fuzz_target_1.rs

```diff
@@ -0,0 +1,8 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|data: &[u8]| {
+    if let Ok(s) = std::str::from_utf8(data) {
+        linkerd_transport_header::fuzz_logic::fuzz_entry(s, data);
```

Comment:
We have a CI task that runs `cargo check` in every directory that has a `Cargo.toml` file and this seems to cause some issue:

```
error[E0433]: failed to resolve: could not find `fuzz_logic` in `linkerd_transport_header`
 --> fuzz_targets/fuzz_target_1.rs:6:35
  |
6 |         linkerd_transport_header::fuzz_logic::fuzz_entry(s, data);
  |                                   ^^^^^^^^^^ could not find `fuzz_logic` in `linkerd_transport_header`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `linkerd-transport-header-fuzz`

To learn more, run the command again with --verbose.
```

Perhaps the whole file needs a `#![cfg(fuzzing)]` at the top so it's only compiled in this mode? Either that, or we should instead use a feature-gate on the transport-header crate and have this fuzzing crate depend on that feature.

Reply:
I solved this by putting a feature hurdle in the fuzzing `Cargo.toml`s